### ISSUE TITLE
NPC range check should not use miner/harvester ranges

### ISSUE THREAD

**Issue by MikeJeffers** [OpenPerpetuum/PerpetuumServer#402]
This query/computation : https://github.com/OpenPerpetuum/PerpetuumServer/blob/Development/src/Perpetuum/Zones/NpcSystem/Npc.cs#L1301-L1304
includes miner/harvester/RSA/remote reps etc.  and that is likely not where the npc needs to limit its range to.
Figure out a better way to filter to the desired set of offensive modules and compute the min range (instead of just `active` and `IsRanged`)

**Comment by Ramit110** [comment]
Not the best implementation but it does work for any weapon that uses ``WeaponModule`` or is a subclass of ``WeaponModule``.

This would break if the class is in someway refactored or modified and I still need to get my local branch working to fully test this.

**Comment by Ramit110** [comment]
Thinking about the current implementation the range of neutralisers/vampires or other non-weapon combat modules won't be included so this code likely won't be the best, I guess I can:
- Add a boolean to ``ActiveModule`` objects that are set true by any modules that want to be included in this Selection. (This one makes the most sense I think)
- Creating a seperate list of modules for ``ActiveCombatModules`` that is iterated through instead.

**Comment by MikeJeffers** [comment]
Yes, good observation.  It might instead be good to do a negation of some known non-combative ranged modules that some npcs are equipped with.
I could see it working either way with your 2 approaches.  I think the property might be good, more efficient to filter on, and could simply be override in the subclasses that you wish to exclude from this range-computation.  But wouldn't worry too much about optimizing the filter as it is cached per npc instance.  Perhaps the property is more maintainable?  Just might need a note that it is used for precisely determining this npc property for those implementing new modules.

**Comment by aqpanaciy** [comment]
I suggest using the [OffensiveModule](https://github.com/OpenPerpetuum/PerpetuumServer/blob/9dbb1c92e6efe7ba070f2da579b0ef980416250d/src/Perpetuum.ExportedTypes/EntityAttributeFlags.cs#L77) property for filtering in this case.

**Comment by Ramit110** [comment]
Just need to test this now

**Comment by aqpanaciy** [comment]
I tested a new version of the filter on industrial NPCs:

- Industrial NPCs no longer snuggle up to the player
- They use energy weapons from a greater distance

 [YouTube](https://youtu.be/ZzpjrczJOhE)

PS: Scarabs do not have weapons, only electronic warfare modules., They still try to keep a line of sight with the player.

**Comment by Ramit110** [comment]
Would that last part be considered a bug?
Otherwise it looks much better!

**Comment by aqpanaciy** [comment]
It doesn't seem to be a bug, it's a feature.
